pkg/resources/validation/cluster/cluster.go

```diff
@@ -93,16 +154,136 @@ func (v *Validator) Admit(request *admission.Request) (*admissionv1.AdmissionRes
+					return psa.AdmissionResponseAllowed(), nil
+				}
+				oldConfig, _ := psa.GetPluginConfigFromCluster(oldCluster)
+				if reflect.DeepEqual(newConfig, oldConfig) {
+					msg := "The Plugin Config for PodSecurity under kube-api.admission_configuration is the same as the previously-set PodSecurityAdmissionConfigurationTemplate." +
+						" Please either change the Plugin Config or set the DefaultPodSecurityAdmissionConfigurationTemplateName."
+					return psa.AdmissionResponseBadRequest(msg), nil
+				}
+			}
+		}
+	}
+	return psa.AdmissionResponseAllowed(), nil
 }
 
-func toExtra(extra map[string]authenticationv1.ExtraValue) map[string]v1.ExtraValue {
-	result := map[string]v1.ExtraValue{}
-	for k, v := range extra {
-		result[k] = v1.ExtraValue(v)
+// checkPSAConfigOnCluster validates the cluster spec when DefaultPodSecurityAdmissionConfigurationTemplateName is set
+func (v *Validator) checkPSAConfigOnCluster(cluster *apisv3.Cluster) (*admissionv1.AdmissionResponse, error) {
+	// validate that extra_args.admission-control-config-file is not set at the same time
+	_, found := cluster.Spec.RancherKubernetesEngineConfig.Services.KubeAPI.ExtraArgs["admission-control-config-file"]
+	if found {
+		msg := "could not use external admission control configuration file when using PodSecurityAdmissionConfigurationTemplate"
+		return psa.AdmissionResponseBadRequest(msg), nil
 	}
-	return result
+	// validate that the configuration for PodSecurityAdmission under the kube-api.admission_configuration section
+	// matches the content of the PodSecurityAdmissionConfigurationTemplate specified in the cluster
+	name := cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+	template, err := v.psact.Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return psa.AdmissionResponseBadRequest(err.Error()), nil
+		}
+		return nil, fmt.Errorf("failed to get PodSecurityAdmissionConfigurationTemplate [%s]: %w", name, err)
+	}
+	fromTemplate, err := psa.GetPluginConfigFromTemplate(template)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the PluginConfig: %w", err)
+	}
+	fromAdmissionConfig, found := psa.GetPluginConfigFromCluster(cluster)
+	if !found {
+		msg := "PodSecurity Configuration is not found under kube-api.admission_configuration"
+		return psa.AdmissionResponseBadRequest(msg), nil
+	}
+	psaConfig := &psav1.PodSecurityConfiguration{}
+	err = json.Unmarshal(fromTemplate.Configuration.Raw, psaConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal PodSecurityConfiguration: %w", err)
+	}
+
+	psaConfig2 := &psav1.PodSecurityConfiguration{}
+	err = json.Unmarshal(fromAdmissionConfig.Configuration.Raw, psaConfig2)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal PodSecurityConfiguration: %w", err)
+	}
+	if !reflect.DeepEqual(psaConfig, psaConfig2) {
+		msg := "PodSecurity Configuration under kube-api.admission_configuration does not match " +
```

Comment:
Same as previous comment - what if PSACT got modified _after_ it was applied to the cluster - we will start hitting this error on any update to the cluster (unless mutating webhook is guaranteed to act first and ensures the contents match?).

Reply:
> unless mutating webhook is guaranteed to act first and ensures the contents match
Yes, this is guaranteed